src/fixed_rational.jl

```diff
@@ -47,14 +55,28 @@ Base.convert(::Type{F}, x::Rational) where {F<:FixedRational} = F(x)
 Base.convert(::Type{Rational{R}}, x::F) where {R,F<:FixedRational} = Rational{R}(x.num, denom(F))
 Base.convert(::Type{Rational}, x::F) where {F<:FixedRational} = Rational{eltype(F)}(x.num, denom(F))
 Base.convert(::Type{AF}, x::F) where {AF<:AbstractFloat,F<:FixedRational} = convert(AF, x.num) / convert(AF, denom(F))
-Base.convert(::Type{I}, x::F) where {I<:Integer,F<:FixedRational} = convert(I, convert(Rational, x))
+Base.convert(::Type{I}, x::F) where {I<:Integer,F<:FixedRational} = isinteger(x) ? div(x.num, denom(F)) : throw(InexactError(:convert, I, x))
 Base.round(::Type{T}, x::F) where {T,F<:FixedRational} = div(convert(T, x.num), convert(T, denom(F)), RoundNearest)
+Base.decompose(x::F) where {T,F<:FixedRational{T}} = (x.num, zero(T), denom(F))
+
+# Promotion rules:
+Base.promote_rule(::Type{<:FixedRational{T1,den1}}, ::Type{<:FixedRational{T2,den2}}) where {T1,T2,den1,den2} = FixedRational{promote_type(T1,T2),den*den2}
```

Comment:
Actually this method should probably be deleted. We don't want users to ever work with two different denominators when using FixedRationals (type explosion!). If they need it they can just use Rational instead which is safer.

Reply:
(Or turned into a helpful error)